src/Core/Domain/ValueObject/Password.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Exception\PasswordConstraintException;
+use PrestaShop\PrestaShop\Core\Security\PasswordPolicyConfiguration;
+use ZxcvbnPhp\Zxcvbn;
+
+/**
+ * Stores plain password
+ */
+class Password
+{
+    /**
+     * @var string
+     */
+    private $password;
+
+    /**
+     * @var int
+     */
+    private $minScore;
+
+    /**
+     * @var int
+     */
+    private $minLength;
+
+    /**
+     * @var int
+     */
+    private $maxLength;
+
+    /**
+     * @param string $password
+     * @param int|null $minLength
+     * @param int|null $maxLength
+     * @param int|null $minScore
+     */
+    public function __construct(string $password, $minLength = null, $maxLength = null, $minScore = null)
```

Comment:
Value objects are not designed to implement such configurable validation, and they also shouldn't have such optional parameters. It means the object is by no mean able to validate itself.

For such complex validations (the complexity here lies in the rules that can be dynamically configured) we need a dedicated validator service. You already created a constraint validator but this one is tightly coupled with the Symfony form framework and won't be usable in other part of the code including in CQRS handlers.

So I suggest creating a core `PasswordValidator` service, in which you can inject:
- the validation configuration about password length
- the Zcbn service which should be injected via DI instead of being created inside the service manually

Then the form constraint validator will be able to use this core service as well.